src/Microsoft.AspNetCore.Authentication.OpenIdConnect/OpenIdConnectHandler.cs

```diff
@@ -919,22 +916,18 @@ private string ReadNonceCookie(string nonce)
 
             foreach (var nonceKey in Request.Cookies.Keys)
             {
-                if (nonceKey.StartsWith(OpenIdConnectDefaults.CookieNoncePrefix))
+                if (nonceKey.StartsWith(Options.NonceCookie.Name))
                 {
                     try
                     {
-                        var nonceDecodedValue = Options.StringDataFormat.Unprotect(nonceKey.Substring(OpenIdConnectDefaults.CookieNoncePrefix.Length, nonceKey.Length - OpenIdConnectDefaults.CookieNoncePrefix.Length));
+                        var nonceDecodedValue = Options.StringDataFormat.Unprotect(nonceKey.Substring(Options.NonceCookie.Name.Length, nonceKey.Length - Options.NonceCookie.Name.Length));
                         if (nonceDecodedValue == nonce)
                         {
-                            var cookieOptions = new CookieOptions
+                            var cookieOptions = Options.NonceCookie.Build(Context);
+                            if (Options.NonceCookie.Path == null)
                             {
-                                HttpOnly = true,
-                                Path = OriginalPathBase + Options.CallbackPath,
-                                SameSite = Http.SameSiteMode.None,
-                                Secure = Request.IsHttps
-                            };
-
-                            Options.ConfigureNonceCookie?.Invoke(Context, cookieOptions);
+                                cookieOptions.Path = OriginalPathBase + Options.CallbackPath;
```

Comment:
Maybe just do this instead? `cookieOptions.Path = cookieOptions.Path ?? OriginalPathBase + Options.CallbackPath;`

There's also a subtle bug here, since even if Path == null, the Builder could set the cookie.Path, which would get blasted away